aep/general/0162/aep.md.j2

```diff
@@ -1,5 +1,396 @@
 # Resource Revisions
 
-**Note:** This AEP has not yet been adopted. See
-[this GitHub issue](https://github.com/aep-dev/aep.dev/issues/16) for more
-information.
+Some APIs need to have resources with a revision history, where users can
+reason about the state of the resource over time. There are several reasons for
+this:
+
+- Users may want to be able to roll back to a previous revision, or diff
+  against a previous revision.
+- An API may create data which is derived in some way from a resource at a
+  given point in time. In these cases, it may be desirable to snapshot the
+  resource for reference later.
+
+**Note:** We use the word _revision_ to refer to a historical reference for a
+particular resource, and intentionally avoid the term _version_, which refers
+to the version of an API as a whole.
+
+## Guidance
+
+APIs **may** store a revision history for a resource. Examples of when it is
+useful include:
+
+- When it is valuable to expose older revisions of a resource via an API. This
+  can avoid the overhead of the customers having to write their own API to store
+  and enable retrieval of revisions.
+- Other resources depend on different revisions of a resource.
+- There is a need to represent the change of a resource over time.
+
+APIs implementing resources with a revision **should** abstract resource
+revisions as nested collection of the resource. Sometimes, the revisions
+collection can be a top level collection. Exceptions include:
+
+- If resource revisions are meant to have longer lifespan than the parent
+resource. In other words, resource revisions exist after resource deletion.
+
+{% tab proto %}
+
+```proto
+message BookRevision {
+  // The path of the book revision.
+  string path = 1;
+
+  // The snapshot of the book
+  Book snapshot = 2
```

Comment:
I'd be fine with `book`. But personally I hate that we landed on the resource singular for the resource, instead of a standard name.

Even for the CRUD methods I'd prefer something more generic, like "resource", rather than the resource singular - it vastly simplifies automation which would have to go extract the resource singular to know what the right field to extract the resource would be.

Reply:
I filed a larger issue on this - I think I'd like to argue we go forward with standardizing and always calling it "snapshot": https://github.com/aep-dev/aep.dev/issues/205

Reply:
I guess there are two topics here:

1. "<resource>" vs. something static
2. Which static thing?

I'm inclined to agree on (1), even though the implications are inconvenient for me/Roblox (extrapolating that we'll conclude the same for e.g. CreateBookRequest).

For (2) though:  why "snapshot" and not "revision" or "resource"?

Reply:
sounds good, focusing on 2 (1 we can discuss in #205): My vote would be for "resource" - then it could even use the same noun as the standard methods, which would be amazingly consistent.